src/Hintman/Hint/NoNewlineAtFileEnd.hs

```diff
@@ -0,0 +1,76 @@
+module Hintman.Hint.NoNewlineAtFileEnd
+       ( getNoNewlineAtFileEndComments
+       ) where
+
+import Hintman.Core.Hint (Hint, HintType (NoNewlineAtFileEnd), Line (..), simpleSuggestion, toLines)
+import Hintman.Core.PrInfo (ModifiedFile (..))
+import Hintman.Core.Review (Comment (..), createComment)
+
+import qualified Data.Text as T
+
+
+-- | Creates 'Comment's on adding newlines in all modified files.
+getNoNewlineAtFileEndComments :: [ModifiedFile] -> [Comment]
+getNoNewlineAtFileEndComments = mapMaybe getFileNoNewlineAtFileEndComment
+
+-- | Takes a file and returns a comment if file doesn't end with a '\n'.
+getFileNoNewlineAtFileEndComment :: ModifiedFile -> Maybe Comment
+getFileNoNewlineAtFileEndComment mf@ModifiedFile{..} = spawnComment $
+    maybeGetNonEmptyLastLine fileLines diffText
+  where
+    diffText :: Text
+    diffText = decodeUtf8 mfContent
+
+    fileLines :: [Line]
+    fileLines = toLines diffText
+
+    hint :: Hint
+    hint = simpleSuggestion NoNewlineAtFileEnd "No newline at end of file" ""
```

Comment:
I just understood that created hints will suggest to replace a non-empty line with an empty one (instead of suggesting the same line with `\n` added), which is not correct. I'll fix this and update the tests soon.

Reply:
Fixed.